BTrees/_base.py

```diff
@@ -815,9 +815,7 @@ def __delitem__(self, key):
         self._del(self._to_key(key))
 
     def clear(self):
-        if self._data:
-            # In the case of __init__, this was already set by __new__
-            self._data = []
+        self._data = []
```

Comment:
I see. I hope my idea about `__keep__` in https://github.com/zopefoundation/persistent/pull/44#issuecomment-256998968 should allow retaining working state and compatibility more or less easily.

I mean for BTree case we could just first add `__keep__ = ['_data'] # or __slots__` and then it works as it was working before. Then we can remove the `__keep__` alongside reworking `__setstate__` to not call clear but directly setting `._data`

I hope this way it could be made that everyone are happy.